### ISSUE TITLE
Rename default branch from `master` to `main`

### ISSUE THREAD

**Issue by mattwthompson** [mdtraj/mdtraj#1862]
This is a shift that happened years ago, so I'm not going to provide more context here. Except that I'm lazy and, on top of the other reasons for doing so, having different default branch names across projects is clunky developer experience.

There'll probably be some tooling changes that need to happen

**Comment by sukritsingh** [comment]
+1 to doing this to be consistent with modern conventions. Might be handy to list out all the tooling that may be affected: 

1. CI
2. Docs/RTD
3. conda feedstock?
4. Pypi?

**Comment by mattwthompson** [comment]
1. Should be handled in the first pass
2. Also should be updated in the first PR
3. Should be unaffected - it just grabs release tarballs
4. Local builds should be unaffected but maybe #1833 would need an update

**Comment by sukritsingh** [comment]
Sounds like it'll just be the CI + Docs mainly then! We can check in with #1833 once we have a PR going probably. I'll try getting to this at some point but if someone else wants to go for it I don't want to be the blocker. My incentive is to use this as an opportunity to learn about the CI setup!

**Comment by janash** [comment]
The current CI runs against both `master` and `main`.  Is there more in the CI that needs to be updated?

@sukritsingh - what would need to be done for the docs?

**Comment by mattwthompson** [comment]
I think we just need to update `main` to be the "default" in the GitHub UI. I'm not sure why this hasn't been done (maybe I just forgot?)

There are a few references to `master` that could be updated: https://github.com/search?q=repo%3Amdtraj%2Fmdtraj%20master&type=code

I'm in favor of making the switch now and cleaning up the mess later

**Comment by janash** [comment]
I'm also in favor of switching now. 

It looks like most of the references to master either aren't referring to a branch or are in comments. Notably, the URLs go to other repositories and are not valid anymore (`404`)

I do suspect that we will need to be done is editing or regenerating `_version.py` to use `main` instead of `master`.

**Comment by Lnaden** [comment]
When you do a branch rename through github, it does a map to the new branch name. So any old bits of code which still reference the "remote" here on github get correctly pointed:

https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/renaming-a-branch

**Comment by janash** [comment]
Yes :) even with that, the URLs in the comments point at `404`s currently. I saw a URL to Gromacs where a file has been renamed, and one to NumPy where the file also appears to have moved. So although the URL in the comment is switching `master` to `main`, the URL still is not pointing where intended.

But internal ones for MDTraj should be fine, yes.

**Comment by mattwthompson** [comment]
Huh, oddly there's not a `main` branch upstream. I have one locally but never pushed it? I guess those CI references didn't do anything.

I'm pulling the trigger, though. I ran this locally:

```
$ git fetch --all                                                                  11:13:52  ☁  c2e0f5a4 ☀
Fetching origin
Fetching jeremy
(interchange-examples-env) [mdtraj]
$ git checkout origin/master                                                       11:13:58  ☁  c2e0f5a4 ☀
HEAD is now at c2e0f5a4 Merge pull request #1951 from rjdirisio/openmm-add-chain
(interchange-examples-env) [mdtraj]
$ git checkout -b main                                                             11:14:46  ☁  c2e0f5a4 ☀
Switched to a new branch 'main'
(interchange-examples-env) [mdtraj]
$ git push origin main                                                                 11:14:48  ☁  main ☀
Total 0 (delta 0), reused 0 (delta 0), pack-reused 0 (from 0)
remote:
remote: Create a pull request for 'main' on GitHub by visiting:
remote:      https://github.com/mdtraj/mdtraj/pull/new/main
remote:
To github.com:mdtraj/mdtraj.git
 * [new branch]        main -> main
```

and made the change in the repo's settings:

![Image](https://github.com/user-attachments/assets/23c1a46a-a145-4563-9e9c-f50a98d37acd)

**Comment by mattwthompson** [comment]
This does mean open PRs need to be updated to point to `main`. I figure most of us have dealt with that but if any external contributors haven't, first click "edit" in the top-right of a PR:

![Image](https://github.com/user-attachments/assets/4cd92abb-7621-492f-91f5-1701aea70152)

and then select "main" in the drop-down here:

![Image](https://github.com/user-attachments/assets/8b70b9a5-9c45-4de5-b6ef-aec63b202636)

**Comment by janash** [comment]
Wow, it's done! Amazing! 🎉
